core/src/main/java/hudson/model/Actionable.java

```diff
@@ -73,13 +73,20 @@
      */
     @Deprecated
 	public List<Action> getActions() {
-		if(actions == null) {
-			synchronized (this) {
-				if(actions == null) {
-					actions = new CopyOnWriteArrayList<Action>();
-				}
-			}
-		}
+        if(actions == null) {
+            synchronized (this) {
+                if (actions == null) {
+                    actions = new CopyOnWriteArrayList<Action>();
+                } else {
+                    for (Action action : actions) {
+                        if (action == null) {
+                            LOGGER.log(Level.WARNING, "Actionable#getActions: Found a null action: " + action.getClass().toString());
+                            actions.remove(action);
```

Comment:
This would throw `ConcurrentModificationException`. Read the docs for `Iterator.remove`.